lib/output/output.go

```diff
@@ -67,8 +67,47 @@ type OutputOpts struct {
 // hook into NewOutput to perform any required setup.
 var newOutputPlatformQuirks func(o *Output) error
 
+// newCapabilityWatcher returns a channel that receives a message when
+// capabilities are updated. By default, no watching functionality is
+// available.
+var newCapabilityWatcher func() chan capabilities = func() chan capabilities { return nil }
+
 func NewOutput(w io.Writer, opts OutputOpts) *Output {
-	caps := detectCapabilities()
+	caps, err := detectCapabilities()
+	o := &Output{caps: overrideCapabilitiesFromOptions(caps, opts), opts: opts, w: w}
+	if newOutputPlatformQuirks != nil {
+		if err := newOutputPlatformQuirks(o); err != nil {
+			o.Verbosef("Error handling platform quirks: %v", err)
+		}
+	}
+
+	// If we got an error earlier, now is where we'll report it to the user.
+	if err != nil {
+		block := o.Block(Linef(EmojiWarning, StyleWarning, "An error was returned when detecting the terminal size and capabilities:"))
+		block.Write("")
+		block.Write(err.Error())
+		block.Write("")
+		block.Write("Execution will continue, but please report this, along with your operating")
+		block.Write("system, terminal, and any other details, to:")
+		block.Write("  https://github.com/sourcegraph/sourcegraph")
```

Comment:
```suggestion
		block.Write("  https://github.com/sourcegraph/sourcegraph/issues/new")
```

for convenience, maybe?

Reply:
Yeah, I debated that. I think you're right.